### PR TITLE
Refactor forward message

### PR DESCRIPTION
**Pull request by Tougee**

### CODE REVIEW

File: app/src/main/java/one/mixin/android/extension/UrlExtension.kt [hunk 1]
```diff
@@ -214,3 +191,56 @@ private fun String.isAppScheme() = startsWith(Constants.Scheme.APPS, true) ||
...
+    showNow: Boolean = true,
+    afterShareText: (() -> Unit)? = null,
+    afterShareData: (() -> Unit)? = null,
+    onError: ((String) -> Unit)? = null,
+) {
+    val text = this.getQueryParameter("text")
+    text.notNullWithElse(
+        {
+            ForwardActivity.show(
+                MixinApplication.appContext,
```
Comment (SeniorZhai) [inline app/src/main/java/one/mixin/android/extension/UrlExtension.kt]: Application context sometime can't start activity

**Review by SeniorZhai — CHANGES_REQUESTED** [review]

File: app/src/main/java/one/mixin/android/util/ShareHelper.kt [hunk 2]
```diff
@@ -20,61 +22,50 @@ class ShareHelper {
...
-    fun generateForwardMessageList(intent: Intent): ArrayList<ForwardMessage>? {
+    fun generateForwardMessageList(intent: Intent): ArrayList<ForwardMessage<ForwardCategory>>? {
         val action = intent.action
         val type = intent.type
         if (action == null || type == null) {
             return null
         }
-        val result = arrayListOf<ForwardMessage>()
+        val result = arrayListOf<ForwardMessage<ForwardCategory>>()
+        // TODO handle */*
```
Comment (SeniorZhai) [inline app/src/main/java/one/mixin/android/util/ShareHelper.kt:32]: Is it done?

**Review by SeniorZhai — CHANGES_REQUESTED** [review]

File: app/src/main/java/one/mixin/android/extension/UrlExtension.kt (hunk 1 above)
Comment (Tougee) [reply app/src/main/java/one/mixin/android/extension/UrlExtension.kt]: done

**Review by SeniorZhai — APPROVED** [review]

File: app/src/main/java/one/mixin/android/util/ShareHelper.kt (hunk 2 above)
Comment (Tougee) [reply app/src/main/java/one/mixin/android/util/ShareHelper.kt:32]: This doesn't affect current behavior and can be improved in the future commits.

**Pull request merged**